### PR TITLE
Extend support for parsing arrays from params hash

### PR DESCRIPTION
**Pull request by lvonk**
This commit also enables parsing array(Integer) and other simple types.
Furthermore the as_params, used mostly in specs, now also creates
similar hashes for posting as your browser would do.

### CODE REVIEW

File: lib/sequent/core/helpers/param_support.rb [hunk 1]
```diff
@@ -45,36 +51,54 @@ def as_params
...
-              value = value.as_params
-            elsif value.kind_of?(Array)
-              value = value.map { |val| val.kind_of?(ValueObject) ? val.as_params : val }
-            elsif value.is_a? DateTime
-              value = value.iso8601
-            elsif value.is_a? Date
-              value = value.strftime("%d-%m-%Y") # TODO Remove to TypeConverter
-            end
-            hash[field[0]] = value
+            hash[field[0]] = if value.kind_of?(Array)
```
Comment (bjpbakker) [inline lib/sequent/core/helpers/param_support.rb]: Should we support any `Enumerable` here, not just `Array`?

**Comment by bjpbakker** [comment]
Nice refactoring :+1:

File: lib/sequent/core/helpers/param_support.rb (hunk 1 above)
Comment (lvonk) [inline lib/sequent/core/helpers/param_support.rb]: I think we eventually should, this would I guess be another PR since we have to look into ArrayWithType and array(Integer) etc.

Comment (bjpbakker) [inline lib/sequent/core/helpers/param_support.rb]: Agreed.
